### PR TITLE
new miRanda module

### PR DESCRIPTION
**Pull request by BarryDigby**
## PR checklist

miRanda miRNA target prediction algorithm. 

- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/miranda/main.nf [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    miranda \\
+        $mirbase \\
+        $fasta \\
+        $args \\
+        -out ${prefix}.out
```
Comment (ramprasadn) [inline modules/nf-core/miranda/main.nf:29]: Shouldn't this be ${prefix}.txt?

Comment (BarryDigby) [reply modules/nf-core/miranda/main.nf:29]: the output from `miRanda` is a mess and needs to be reformatted, so this `.out` file is temporary. 

The code below it is present in virtually all `miRanda` workflows, definitely worthwhile including it in the module :)

File: modules/nf-core/miranda/main.nf [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::miranda=3.3a" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/miranda:3.3a--h779adbc_3':
+        'quay.io/biocontainers/miranda:3.3a--h779adbc_3' }"
+
+    input:
+    tuple val(meta), path(fasta)
```
Comment (matthdsm) [inline modules/nf-core/miranda/main.nf:11]: I'm not very familiar with these applications, but is there any other name you could give this input? 
`fasta` is generally used for reference genome fasta files and could cause some confusion?

**Review by nvnieuwk — APPROVED** [review]
LGTM

**Review by matthdsm — APPROVED** [review]

**Pull request merged**